File: src/main/java/green/seagull/very/good/purchase/domain/Purchase.java
package green.seagull.very.good.purchase.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDate;

@Document
public class Purchase {
    @Id
    private String id;

    private String name;
    private LocalDate date;
    private PurchaseType purchaseType;
    private BigDecimal amountDollars;

    public Purchase(String name, LocalDate date, PurchaseType purchaseType, BigDecimal amountDollars) {
        this.name = name;
        this.date = date;
        this.purchaseType = purchaseType;
        this.amountDollars = amountDollars;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public LocalDate getDate() {
        return date;
    }

    public PurchaseType getPurchaseType() {
        return purchaseType;
    }

    public BigDecimal getAmountDollars() {
        return amountDollars;
    }
}

File: src/main/kotlin/green/seagull/very/good/purchase/domain/Purchase.kt
package green.seagull.very.good.purchase.domain

import green.seagull.very.good.purchase.model.PurchaseType
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigDecimal
import java.time.LocalDate

@Document
data class Purchase(
    @Id var id: String? = null,
    var name: String,
    var date: LocalDate,
    var purchaseType: PurchaseType,
    var amountDollars: BigDecimal
)
